### PR TITLE
feat: Add check API

### PR DESCRIPTION
**Pull request by kelwang**
Closes https://github.com/influxdata/influxdb/issues/14325

add check struct, kv, http, authorizer, influxd launcher and testings. based on the swagger file design.

- [ ] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass
- [x] http/swagger.yml updated (if modified Go structs or API)
- [x] Documentation updated or issue created (provide link to issue/pr)
- [ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: notification/check/check.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+func (b *Base) SetDescription(description string) {
+	b.Description = description
+}
+
+// SetStatus implements influxdb.Updator interface.
+func (b *Base) SetStatus(status influxdb.Status) {
+	b.Status = status
+}
+
+var typToCheck = map[string](func() influxdb.Check){
```
Comment (ebb-tide) [inline notification/check/check.go]: typToCheck looks like a typo?

File: notification/check/check.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+			Msg:  "Check OrgID is invalid",
+		}
+	}
+	if b.Status != influxdb.Active && b.Status != influxdb.Inactive {
+		return &influxdb.Error{
+			Code: influxdb.EInvalid,
+			Msg:  "invalid status",
+		}
+	}
+	if b.StatusMessageTemplate == "" {
```
Comment (ebb-tide) [inline notification/check/check.go]: Status message template can be empty.

File: check.go [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+// CheckFilter represents a set of filters that restrict the returned results.
+type CheckFilter struct {
+	ID    *ID
+	Name  *string
+	OrgID *ID
+	Org   *string
+}
+
+// QueryParams Converts NotificationRuleFilter fields to url query params.
```
Comment (ebb-tide) [inline check.go]: Check Filter instead of NotificationRuleFilter here.

File: kv/check.go [hunk 4]
```diff
@@ -0,0 +1,704 @@
...
+			return nil, &influxdb.Error{
+				Err: err,
+			}
+		}
+		return c, nil
+	}
+
+	err = s.kv.View(ctx, func(tx Tx) error {
+		if filter.Org != nil {
+			o, err := s.findOrganizationByName(ctx, tx, *filter.Org)
```
Comment (ebb-tide) [inline kv/check.go]: line 131 suggests that orgID lookup will be efficient but it looks like this function searches for org Name only?

Comment (kelwang) [reply kv/check.go]: Good point, I think it's related the org and orgID filter issue, we will make update for all them in one pr.

**Pull request merged**